src/support.h

```diff
@@ -29,16 +32,44 @@ class AreaSupport {
      */
     static void generateSupportAreas(SliceDataStorage& storage, unsigned int mesh_idx, unsigned int layer_count, std::vector<Polygons>& supportAreas);
 
+    /*!
+     * Generate support bottom areas for a given mesh.
+     *
```

Comment:
It's not clear from the description what happens when you call generateBottoms before generateRoofs or the other way around.

It should be explicitly noted where bottoms are generated and how : above the model and within the support areas.